frontend-ssr/redux/exercise-submissions/exercise-submissions.reducer.ts

```diff
@@ -0,0 +1,54 @@
+import { SubmissionState } from './types';
+import { EXERCISE_SUBMISSIONS } from './exercise-submissions.actions';
+
+const initialState: SubmissionState = {
+  submissions: [],
+};
+export const submissionReducer = (state = initialState, action: { type: string; payload: any;})
+: SubmissionState => {
+  switch (action.type) {
+    case EXERCISE_SUBMISSIONS.ADD: {
+      let { index, submission } = action.payload;
+      if (Array.isArray(submission)) {
+        submission = [...submission];
+      } else {
+        submission = [submission];
+      }
+      return {
+        submissions: [...state.submissions.slice(0, index),
+          ...submission,
+          ...state.submissions.slice(index)],
+      };
+    }
+    case EXERCISE_SUBMISSIONS.REMOVE: {
+      let { id } = action.payload;
+      if (Array.isArray(id)) {
```

Comment:
We can apply the same approach as above, if it's not an Array treat it as such. Then we will have just one return statement.